calc/lib/helpers.ex

```diff
@@ -0,0 +1,77 @@
+defmodule Helpers do
+  @moduledoc """
+  Helpers module for parsing and converting data purposes
+  """
+
+  @allowed_parsed_elements 3
+
+  @doc """
+  Initial User input parsing with regular expression
+
+  ## Examples
+      iex> Helpers.parse_input("1+2")
+      {:ok, {"1", "+", "2"}}
+  """
+  @spec parse_input(String.t()) :: {:ok, tuple()} | {:error, String.t()}
+  def parse_input(input) do
+    parsed_input = Regex.scan(~r{\d+\.?\d+|\d+|[+*/-]|[\w]+}, input) |> List.flatten() |> List.to_tuple()
+
+    case tuple_size(parsed_input) do
+      @allowed_parsed_elements -> {:ok, parsed_input}
+      _ -> {:error, "incorrect input has been provided"}
+    end
```

Comment:
Матчить можно прямо на туплу, или список. И, кажется, вы потеряли числа со знаком.
```suggestion
    ~r{[+-]?\d+(?:\.\d+)?|[+*/-]|\w+}
    |> Regex.scan(input, capture: :first)
    |> case do
      [[num1], [sign], [num2]] -> {:ok, {num1, sign, num2}}
      _ -> {:error, "incorrect input has been provided"}
    end
```

Reply:
> Матчить можно прямо на туплу, или список. И, кажется, вы потеряли числа со знаком.

Так выглядит намного приятнее!

Reply:
В добавок, поправил регулярку, чтобы можно было принимать выражение типа `1+2` (слитно)